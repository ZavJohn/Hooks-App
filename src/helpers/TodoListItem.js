import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleToggle, index }) => {
  return (
    <li
        key = { todo.id }
        className="list-group-item"
    > 
        <p className={`text-start ${ todo.done && 'complete'}`}
            onClick={ () => handleToggle ( todo.id )}
        >
                { index + 1 }.{ todo.desc }
            </p>
        <div className="d-grid gap-2 d-md-flex justify-content-md-end">
            <button
                className="btn btn-danger btn-sm"
                onClick={ () => handleDelete( todo.id ) }
            >
                Borrar
            </button>
        </div>
    </li>
  )
}
