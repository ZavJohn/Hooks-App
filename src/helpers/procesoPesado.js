export const procesoPesado = ( iteraciones ) => {

    for ( let i = 0; i < iteraciones; i ++){

        console.log('Inicia el proceso pesado');
    }

    return `${ iteraciones } iteraciones realizadas`;
}