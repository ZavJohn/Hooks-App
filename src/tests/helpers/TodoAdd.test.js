
import React from 'react';
import { shallow } from 'enzyme';
import { TodoAdd } from '../../helpers/TodoAdd';
import '@testing-library/jest-dom';


describe('Pruebas del <TodoAdd/>', () => {

    const handleAddTodo = jest.fn();
    
    const wrapper = shallow(
        <TodoAdd
            handleAddTodo= { handleAddTodo }
        />
        )

    test('Debe mostrarse correctamente', () => { 

        expect( wrapper ).toMatchSnapshot();
     });

     test('No debe llamar el handleAddTodo', () => { 

        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault () {} });

        expect( handleAddTodo).toHaveBeenCalledTimes(0);
     });

     test('Debe llamar el handleAddTodo', () => { 

        const value = 'Aprender react';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });
    
        const formSubmit =wrapper.find('form').prop('onSubmit')
        formSubmit({ preventDefault (){} });
        expect( handleAddTodo ).toHaveBeenCalledTimes(1);
        expect( handleAddTodo ).toHaveBeenLastCalledWith( expect.any(Object) );
        expect( handleAddTodo ).toHaveBeenLastCalledWith({
            desc: value,
            done: false,
            id: expect.any(Number)
        });

        expect( wrapper.find('input').prop('value') ).toBe('');
     });


});