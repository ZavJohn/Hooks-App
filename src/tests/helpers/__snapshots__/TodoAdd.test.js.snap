// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <TodoAdd/> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      complete="false"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2 d-md-flex justify-content-md-end mt-1"
    >
      <button
        className="btn btn-success btn-sm"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</Fragment>
`;
